By default, the boost test option is mandatory for build environment
detection. 
This patch suppress this dependency and allows the test part only if
build_test is enabled at the Buildroot level.

Signed-off-by: Gwenhael Goavec-Merou <gwenhael.goavec-merou@trabucayre.com>
---
Index: gnuradio-3.7.5/volk/cmake/VolkBoost.cmake
===================================================================
--- gnuradio-3.7.5.orig/volk/cmake/Modules/VolkBoost.cmake
+++ gnuradio-3.7.5/volk/cmake/Modules/VolkBoost.cmake
@@ -29,7 +29,6 @@ set(__INCLUDED_VOLK_BOOST_CMAKE TRUE)
 set(BOOST_REQUIRED_COMPONENTS
     filesystem
     system
-    unit_test_framework
     program_options
 )
 
Index: gnuradio-3.7.5/volk/lib/CMakeLists.txt
===================================================================
--- gnuradio-3.7.5.orig/volk/lib/CMakeLists.txt
+++ gnuradio-3.7.5/volk/lib/CMakeLists.txt
@@ -580,7 +580,10 @@
 ########################################################################
 # Build the QA test application
 ########################################################################
-if(ENABLE_TESTING)
+
+find_package(Boost "1.35" COMPONENTS "unit_test_framework")
+
+if(ENABLE_TESTING AND BUILD_TEST)
 
     #include Boost headers
     include_directories(${Boost_INCLUDE_DIRS})
@@ -597,4 +600,4 @@
         TARGET_DEPS volk
     )
 
-endif(ENABLE_TESTING)
+endif(ENABLE_TESTING AND BUILD_TEST)
